File: VB/WindowsApplication1/MyMemoEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Public Class MyMemoEditViewInfo
		Inherits MemoEditViewInfo
		Private Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Public Shadows ReadOnly Property Item() As RepositoryItemMyMemoEdit
			Get
				Return TryCast(MyBase.Item, RepositoryItemMyMemoEdit)
			End Get
		End Property
		Public Shadows ReadOnly Property OwnerControl() As MyMemoEdit
			Get
				Return TryCast(MyBase.OwnerControl, MyMemoEdit)
			End Get
		End Property

		Protected Overrides Sub CalcContentRect(ByVal bounds As Rectangle)
			MyBase.CalcContentRect(bounds)
			If OwnerControl IsNot Nothing Then
				If OwnerControl.VScrollVisible Then
					fContentRect.Width = ContentRect.Width - Item.ScrollWidth + SystemInformation.VerticalScrollBarWidth + 1
				End If
				If OwnerControl.HScrollVisible Then
					fContentRect.Height = ContentRect.Height - Item.ScrollWidth + SystemInformation.VerticalScrollBarWidth + 1
				End If
				fMaskBoxRect = ContentRect
			End If
		End Sub
	End Class

End Namespace

File: CS/WindowsApplication1/MyMemoEditViewInfo.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public class MyMemoEditViewInfo : MemoEditViewInfo
    {
        MyMemoEditViewInfo(RepositoryItem item) : base(item) { }

        public new RepositoryItemMyMemoEdit Item { get { return base.Item as RepositoryItemMyMemoEdit; } }
        public new MyMemoEdit OwnerControl { get { return base.OwnerControl as MyMemoEdit; } }

        protected override void CalcContentRect(Rectangle bounds)
        {
            base.CalcContentRect(bounds);
            if (OwnerControl != null)
            {
                if (OwnerControl.VScrollVisible)
                    fContentRect.Width = ContentRect.Width - Item.ScrollWidth + SystemInformation.VerticalScrollBarWidth + 1;
                if (OwnerControl.HScrollVisible)
                    fContentRect.Height = ContentRect.Height - Item.ScrollWidth + SystemInformation.VerticalScrollBarWidth + 1;
                fMaskBoxRect = ContentRect;
            }
        }
    }

}
